### PR TITLE
feat: Telemetry logged in analytics.testkube.io debug server, Telemetry instead of analytics (rename) 

### PR DESCRIPTION
**Pull request by exu**
## Pull request description 

Missing part - enable/disable server analytics through API on cluster.
Breaking analytics -> telemetry refactor
Events changed need to adapt data on charts

## Checklist (choose whats happened)

- [x] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: internal/app/api/v1/server.go [hunk 1]
```diff
@@ -256,12 +254,30 @@ func (s TestkubeAPI) Init() {
...
 	s.Mux.Mount("/results", s.Mux)
 
-	s.Log.Infow("Testkube API configured", "namespace", s.Namespace, "clusterId", s.ClusterID)
+	s.Log.Infow("Testkube API configured", "namespace", s.Namespace, "clusterId", s.ClusterID, "telemetry", s.TelemetryEnabled)
+}
+
+func (s TestkubeAPI) StartHeartbeat() {
+	go func() {
+		for range time.Tick(HeartbeatInterval) {
+			host, _ := os.Hostname()
```
Comment (vsukhin) [inline internal/app/api/v1/server.go:271]: I prefer to log errors in any case

File: internal/pkg/api/repository/config/mongo.go [hunk 2]
```diff
@@ -31,14 +31,20 @@ func (r *MongoRepository) GetUniqueClusterId(ctx context.Context) (clusterId str
...
-		config.ClusterId = fmt.Sprintf("cluster%s", analytics.MachineID())
+		config.ClusterId = fmt.Sprintf("cluster%s", telemetry.MachineID())
 		err := r.Upsert(ctx, config)
 		return config.ClusterId, err
 	}
 
 	return config.ClusterId, nil
 }
 
+func (r *MongoRepository) GetTelemetryEnabled(ctx context.Context) (ok bool, err error) {
```
Comment (vsukhin) [inline internal/pkg/api/repository/config/mongo.go:42]: Shoud we store it in mongodb? Might be just use ConfigMap?

File: internal/app/api/v1/server.go [hunk 3]
```diff
@@ -256,12 +254,30 @@ func (s TestkubeAPI) Init() {
...
 
 	// mount everything on results
 	// TODO it should be named /api/ + dashboard refactor
 	s.Mux.Mount("/results", s.Mux)
 
-	s.Log.Infow("Testkube API configured", "namespace", s.Namespace, "clusterId", s.ClusterID)
+	s.Log.Infow("Testkube API configured", "namespace", s.Namespace, "clusterId", s.ClusterID, "telemetry", s.TelemetryEnabled)
+}
+
+func (s TestkubeAPI) StartHeartbeat() {
```
Comment (vsukhin) [inline internal/app/api/v1/server.go:268]: Should it be public?

File: pkg/envs/bool.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package envs
+
+import (
+	"os"
+	"strconv"
+)
+
+func IsTrue(name string) (is bool) {
+	if val, ok := os.LookupEnv(name); ok {
+		is, _ = strconv.ParseBool(val)
```
Comment (vsukhin) [inline pkg/envs/bool.go:10]: I prefer to log all errors. Never know what issue you may face

File: pkg/telemetry/analytics.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	return SendData(senders, payload)
+}
+
+// SendHeartbeatEvent will send CLI event to GA
+func SendHeartbeatEvent(host, version, clusterId string) (string, error) {
+	payload := NewAPIPayload(clusterId, "testkube_api_heartbeat", version, host)
+	return SendData(senders, payload)
+}
+
+// SendData sends data to all telemetry storages  in parallel and syncs sending
```
Comment (vsukhin) [inline pkg/telemetry/analytics.go:56]: Is this public method?

File: pkg/telemetry/generators.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	"encoding/hex"
+	"os"
+
+	"github.com/denisbrodbeck/machineid"
+)
+
+// MachineID returns unique user machine ID
+func MachineID() string {
+	id, err := machineid.ProtectedID("testkube")
+	// fallback to hostname based machine id in case of error
```
Comment (vsukhin) [inline pkg/telemetry/generators.go:14]: Again, I would log error somewhere

File: pkg/telemetry/generators.go [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"os"
+
+	"github.com/denisbrodbeck/machineid"
+)
+
+// MachineID returns unique user machine ID
+func MachineID() string {
```
Comment (vsukhin) [inline pkg/telemetry/generators.go:12]: GetMachineID?

**Review by vsukhin — APPROVED** [review]
looks good, some minor notes

File: pkg/envs/bool.go (hunk 4 above)
Comment (exu) [reply pkg/envs/bool.go:10]: logged to testkubes default logger

File: pkg/telemetry/generators.go (hunk 7 above)
Comment (exu) [reply pkg/telemetry/generators.go:12]: done

File: pkg/telemetry/generators.go (hunk 6 above)
Comment (exu) [reply pkg/telemetry/generators.go:14]: logged to testkube default logger

File: internal/pkg/api/repository/config/mongo.go (hunk 2 above)
Comment (exu) [reply internal/pkg/api/repository/config/mongo.go:42]: Yeah. Was thinking about it but need to research problems with access from several nodes at once first. Will add task about it

File: internal/app/api/v1/server.go (hunk 3 above)
Comment (exu) [reply internal/app/api/v1/server.go:268]: no it shouldn't: fixed

**Comment by exu** [comment]
> looks good, some minor notes

@vsukhin extracted one comment into separate issue as it'll be quite big with all testings

**Pull request merged**